distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso

```diff
@@ -343,6 +359,43 @@ java_instance_check value typ =
 is_an : Any -> Any -> Boolean
 is_an value typ = is_a value typ
 
+## UNSTABLE
+   ADVANCED
+
+   Returns the type of the given value.
+
+   Arguments:
+   - value: the value to get the type of.
+type_of : Any -> Any
+type_of value =
+    if is_error value then Base.Error else
+        case value of
+            Integer -> Integer
+            Decimal -> Decimal
+            Number  -> Number
+            _ ->
+                Meta.type_of_builtin value
+
+## UNSTABLE
+   ADVANCED
+   PRIVATE
+
+   Checks if the given type agrees with the expected one
+
+   Arguments:
+   - tpe: the type to check
+   - expected: the expected type
+is_of_type : Any -> Any -> Boolean
+is_of_type tpe expected = @Builtin_Method "Meta.is_of_type_builtin"
```

Comment:
Equality check, no subtyping check at this point. 
It's all fine when you are executing the happy path e.g., comparing values or comparing just types. But when you start comparing values and types (and it is not so hard if you mix `Vector` instead of `Vector.Vector`) then it crashes in an awful way.
I wanted to keep this self-contained, hence a new method, but looks like I will have to see if `==` can be tweaked to make this happen.

Reply:
A slight suggestion that this may be a decent moment to move `==` to be a builtin instead.

Reply:
Yeah, I have a gift for stumbling upon tickets that initially look simple and then I open pandora's box.

Reply:
Delayed a move to `==` due to ongoing statics work and known problems with `==` on type.